fe2-android/app/src/main/java/com/github/dedis/popstellar/model/objects/Channel.java

```diff
@@ -0,0 +1,132 @@
+package com.github.dedis.popstellar.model.objects;
+
+import androidx.annotation.NonNull;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * This class represent a channel used to subscribe to a certain part of the messages. More detailed
+ * information are given in the protocol specification of the application.
+ *
+ * <p>It is implemented in a functional manner to avoid any unintended modification
+ */
+public class Channel {
+
+  private static final String ROOT_CHANNEL = "/root";
+  private static final String DELIMITER = "/";
+  // Predicate matching a String with the protocol regex of a channel
+  private static final Predicate<String> CHANNEL_VALIDATOR =
+      Pattern.compile("^/root(/[^/]+)*$").asPredicate();
+
+  /** Root channel, every client is subscribed to it. */
+  public static final Channel ROOT = new Channel();
+
+  public static Channel newChannel(String value) {
+    // Avoid creating duplicate instances
+    if (ROOT_CHANNEL.equals(value)) return ROOT;
+
+    // Remove the "/root/" part of the value as it is not relevant and does not need to be stored
+    String relevantPart = value.substring(ROOT_CHANNEL.length() + 1);
+    return new Channel(relevantPart.split(DELIMITER));
+  }
+
+  // ================ Object itself =================
+
+  // This list contains the different parts of the channel except the root as every channel has a
+  // root part
+  // So for the channel /root/lao_id/election_id, the elements will be [lao_id, election_id]
+  @NonNull private final List<String> elements;
+
+  private Channel(String... elements) {
+    this.elements = Arrays.asList(elements);
+
+    if (!CHANNEL_VALIDATOR.test(getAsString()))
+      throw new IllegalArgumentException(getAsString() + " is not a valid channel");
+  }
+
+  private Channel(Channel base, String subElement) {
+    List<String> newElems = new ArrayList<>(base.elements);
+    newElems.add(subElement);
+    this.elements = Collections.unmodifiableList(newElems);
+
+    if (!CHANNEL_VALIDATOR.test(getAsString()))
+      throw new IllegalArgumentException(getAsString() + " is not a valid channel");
+  }
+
+  /**
+   * Return a channel that is this channel with one more element underneath
+   *
+   * @param subElement element to append to this channel
+   * @return a new Channel object
+   */
+  public Channel sub(String subElement) {
```

Comment:
I know anything is going to be more verbose, but this is not very explicit. 
How about the following ? or do you have a better idea ? I don't, but I'm not very happy my idea either.
```suggestion
  public Channel getSubChannel(String subElement) {
```

Reply:
I'd go with `subChannel()`.

I went for a brief function name because I wanted this call to be easy to chain :
Social Media channel :
`lao.getChannel().sub("social").sub(token)`

subChannel is more meaningful and should not be too long.